File: Matrix Audio/matrix.audio.client/src/app/models/payment-request.ts
import { Order } from "./order";

export interface PaymentRequest {
    amount: number;
    currency: string;
    description: string;
}

export interface PaymentConfirmRequest {
    tokenId: string;
    order: Order;
}

export interface PaymentResult {
    isSuccess: boolean;
    transactionId?: string;
    errorMessage?: string;
}

File: Matrix Audio/Matrix.Audio.Models/PaymentRequest.cs
namespace Matrix.Audio.Models;

public class PaymentRequest
{
    public decimal Amount { get; set; }
    public string Currency { get; set; } = "usd";
    public string Description { get; set; } = "";
    public List<string> PaymentMethodTypes { get; set; } = ["card"];
}


public class PaymentConfirmRequest : PaymentRequest
{
    public required string TokenId { get; set; }

    public required Order Order { get; set; }
}